### ISSUE TITLE
`Crypto.Random` and forking the `DRG`

### ISSUE THREAD

**Issue by ibotty** [haskell-crypto/cryptonite#15]
Am I right, that forking the DRG is just not possible with every `DRG` in a pure way (i.e. without mixing entropy or using new entropy)?

If so, can you comment on how you envision it to be done? Maybe provide the building blocks to "purely" (but not referential transparent) mix entropy to get two generators or (better) provide a function to do that.

I might not have thought this out though.

**Comment by vincenthz** [comment]
yes, it's not currently possible for the DRG class itself to fork/split.

You could very easily have a main "seed" generator that would seed temporary generators. i.e. something like this:

``` haskell
let (newDRG, seedGenerator') = withDRG seedGenerator drgNew
```

If we had a generic 'drgNew' for the DRG class, then you'ld be able to do that for every DRG.
Would that be enough to cover your cases ?

**Comment by ibotty** [comment]
Yes, a `drgNew` (or `newDRG`) would be enough.

That said, after porting the code that triggered that issue to use the 
`MonadRandom` class, I don't seem to need it anymore.
